pkg/console/operator/sync_v400.go

```diff
@@ -392,6 +403,86 @@ func (co *consoleOperator) SyncRoute(operatorConfig *operatorv1.Console) (*route
 	return rt, rtIsNew, rtErr
 }
 
+func (c *consoleOperator) SyncCustomLogoConfigMap(operatorConfig *operatorsv1.Console) error {
+	// validate first, to avoid a broken volume mount & a crashlooping console
+	okToMount, err := c.ValidateCustomLogo(operatorConfig)
+
+	if okToMount {
+		if err := c.UpdateCustomLogoSyncSource(operatorConfig); err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+// on each pass of the operator sync loop, we need to check the
+// operator config for a custom logo.  If this has been set, then
+// we notify the resourceSyncer that it needs to start watching this
+// configmap in its own sync loop.  Note that the resourceSyncer's actual
+// sync loop will run later.  Our operator is waiting to receive
+// the copied configmap into the console namespace for a future
+// sync loop to mount into the console deployment.
+func (c *consoleOperator) UpdateCustomLogoSyncSource(operatorConfig *operatorsv1.Console) error {
+	source := resourcesynccontroller.ResourceLocation{}
+	logoConfigMapName := operatorConfig.Spec.Customization.CustomLogoFile.Name
+
+	if logoConfigMapName != "" {
+		source.Name = logoConfigMapName
+		source.Namespace = api.OpenShiftConfigNamespace
+	}
+	// if no custom logo provided, sync an empty source to delete
+	return c.resourceSyncer.SyncConfigMap(
+		resourcesynccontroller.ResourceLocation{Namespace: api.OpenShiftConsoleNamespace, Name: api.OpenShiftCustomLogoConfigMapName},
+		source,
+	)
+	return nil
+}
+
+func (c *consoleOperator) ValidateCustomLogo(operatorConfig *operatorsv1.Console) (okToMount bool, invalid error) {
+	logoConfigMapName := operatorConfig.Spec.Customization.CustomLogoFile.Name
+	logoImageKey := operatorConfig.Spec.Customization.CustomLogoFile.Key
+
+	// if unset, allow the sync to set a null source
+	if len(logoConfigMapName) == 0 && len(logoImageKey) == 0 {
+		return true, nil
+	}
+
+	// both set, or both unset
+	if (len(logoConfigMapName) == 0) != (len(logoImageKey) == 0) {
```

Comment:
Isn't this statement an `else` branch of the previous statement ?

Reply:
"Avoid else, return early" seems to be the go mantra, so I went with this form.  I do think it makes logic easier to follow.